include/grpcpp/impl/codegen/sync_stream.h

```diff
@@ -325,6 +325,8 @@ class ClientWriter : public ClientWriterInterface<W> {
     if (options.is_last_message()) {
       options.set_buffer_hint();
       ops.ClientSendClose();
+    } else {
+      options.clear_buffer_hint();  // not supported for sync API
```

Comment:
It's nice to have this comment here, but it would be better if this was in a more obvious place for users. Maybe above Write in line 116? And once you do that, maybe you can remove these comments altogether.

Also the comment is a little deceptive: a reader without any context might think that "clear_buffer_hint" is not supported for sync API. I'd suggest "set_buffer_hint, except for last message, not supported for sync API"; then again, you won't need to address this at all if the above suggestion is addressed.